### PR TITLE
evdev: close file descriptors in a separate thread

### PR DESCRIPTION
**Pull request by Tilka**
For some reason Linux is surprisingly slow at closing file descriptors of event devices. This commit improves GUI startup times on my computer by about 1.5 seconds.

### CODE REVIEW

File: Source/Core/InputCommon/ControllerInterface/evdev/evdev.cpp [hunk 1]
```diff
@@ -665,7 +693,7 @@ evdevDevice::~evdevDevice()
   {
     m_input_backend.RemoveDevnodeObject(node.devnode);
     libevdev_free(node.device);
-    close(node.fd);
+    m_input_backend.DeferredClose(node.fd);
```
Comment (jordan-woyak) [inline Source/Core/InputCommon/ControllerInterface/evdev/evdev.cpp:696]: Are these `fd`s going to remain open until `CleanUp` happens to be called in `PopulateDevices` or immediately after `AddDeviceNode`? They could even leak on shutdown, unless I am missing something? Maybe `CleanUp` should be called every iteration of the `HotplugThreadFunc` loop?

File: Source/Core/InputCommon/ControllerInterface/evdev/evdev.cpp [hunk 2]
```diff
@@ -44,6 +45,30 @@ class InputBackend final : public ciface::InputBackend
...
+    std::vector<int> work;
+    m_cleanup_queue.swap(work);
+    std::thread(
+        [](std::vector<int> vec) {
+          Common::SetCurrentThreadName("evdev cleanup");
+          for (int fd : vec)
+            close(fd);
+        },
+        std::move(work))
+        .detach();
```
Comment (jordan-woyak) [inline Source/Core/InputCommon/ControllerInterface/evdev/evdev.cpp:67]: `detach`ing threads is ugly. You could replace this with our `WorkQueueThread`.

**Review by jordan-woyak — CHANGES_REQUESTED** [review]

File: Source/Core/InputCommon/ControllerInterface/evdev/evdev.cpp (hunk 2 above)
Comment (Tilka) [reply Source/Core/InputCommon/ControllerInterface/evdev/evdev.cpp]: Good idea, done.

File: Source/Core/InputCommon/ControllerInterface/evdev/evdev.cpp (hunk 1 above)
Comment (Tilka) [reply Source/Core/InputCommon/ControllerInterface/evdev/evdev.cpp]: No longer deferred in the latest code.

File: Source/Core/InputCommon/ControllerInterface/evdev/evdev.cpp [hunk 3]
```diff
@@ -8,6 +8,7 @@
 #include <map>
 #include <memory>
 #include <string>
+#include <vector>
```
Comment (jordan-woyak) [inline Source/Core/InputCommon/ControllerInterface/evdev/evdev.cpp:11]: It doesn't look like vector is being used?

**Review by jordan-woyak — APPROVED** [review]
LGTM
